src/GraphQL/ResolveFieldContext/IResolveFieldContext.cs

```diff
@@ -39,6 +39,16 @@ public interface IResolveFieldContext : IProvideUserContext
         /// </summary>
         IDictionary<string, ArgumentValue>? Arguments { get; }
 
+        //TODO: change comments or/and add new methods
+
+        /// <summary>
+        /// A dictionary of directives with their arguments passed to the field, or <see langword="null"/> if no directives were defined for the field.
+        /// It is recommended to use the <see cref="ResolveFieldContextExtensions.GetArgument{TType}(IResolveFieldContext, string, TType)">GetArgument</see>
```

Comment:
Question 4: Add methods to get directive arguments as well? I guess yes. So what name to choose?

Reply:
Have you thought of returning a `DirectiveInfo` or similar class like this?

```cs
class DirectiveInfo : Dictionary<string, ArgumentValue>
{
    public DirectiveDefinition DirectiveDefinition { get; }
    public object GetArgument(...);
}
```

Then on `IResolveFieldContext` we have:

```cs
IDictionary<string, DirectiveInfo>? Directives { get; }
DirectiveInfo? GetDirective(string directiveName);
```

Now you can write code like this:

```cs
if (context.GetDirective("include")?.GetArgument<bool>("if") ?? false)
{
    // code
}
```

And the names are all run through the name converter.

Reply:
> And the names are all run through the name converter.

I think no. NameCoverter works for fields and fields arguments. Directive has no fields and directive is not a graph type.

Reply:
Thanks, done.